examples/llama2/notebook/llama2_multiep/user_script.py

```diff
@@ -0,0 +1,279 @@
+            new_v[:batch_size, :num_heads, :past_seq_len, :head_size] = v
+            ort_inputs[k] = new_v
+    return ort_inputs
+
+
+# -----------------------------------------------------------------------------
+# Conversion Arguments (Inputs, Outputs, Dynamic Axes)
+# -----------------------------------------------------------------------------
+
+
+def get_merged_model_dynamic_axes(input_names: List[str], output_names: List[str]):
+    dynamic_axes = {}
+    for name in input_names + output_names:
+        if name in {"input_ids", "position_ids"}:
+            # shape is (batch_size, sequence_length)
+            dynamic_axes[name] = {0: "batch_size", 1: "sequence_length"}
+        elif name == "attention_mask":
+            # shape is (batch_size, past_sequence_length + sequence_length) = (batch_size, total_sequence_length)
+            # for prompt generation, past_sequence_length = 0
+            # for token generation, sequence_length = 1
+            dynamic_axes[name] = {0: "batch_size", 1: "total_sequence_length"}
+        elif "past" in name:
+            # shape is (batch_size, num_heads, past_sequence_length, head_size)
+            dynamic_axes[name] = {0: "batch_size", 2: "past_sequence_length"}
+        elif name == "logits":
+            # shape is (batch_size, sequence_length, vocab_size)
+            dynamic_axes[name] = {0: "batch_size", 1: "sequence_length"}
+        elif "present" in name:
+            # shape is (batch_size, num_heads, past_sequence_length + sequence_length, head_size)
+            #  = (batch_size, num_heads, total_sequence_length, head_size)
+            # for prompt generation, past_sequence_length = 0
+            # for token generation, sequence_length = 1
+            dynamic_axes[name] = {0: "batch_size", 2: "total_sequence_length"}
+        else:
+            raise ValueError("Unknown input or output name found")
+    return dynamic_axes
+
+
+def get_merged_decoder_with_past_io_config(model: PyTorchModelHandler):
+    config = model.get_hf_model_config()
+
+    input_names = [
+        "input_ids",
+        "attention_mask",
+        "position_ids",
+        *list(
+            chain.from_iterable(
+                (f"past_key_values.{i}.key", f"past_key_values.{i}.value") for i in range(config.num_hidden_layers)
+            )
+        ),
+    ]
+    output_names = [
+        "logits",
+        *list(chain.from_iterable((f"present.{i}.key", f"present.{i}.value") for i in range(config.num_hidden_layers))),
+    ]
+    dynamic_axes = get_merged_model_dynamic_axes(input_names, output_names)
+    return {
+        "input_names": input_names,
+        "dynamic_axes": dynamic_axes,
+        "output_names": output_names,
+    }
+
+
+# -----------------------------------------------------------------------------
+# Metric Data Loader
+# -----------------------------------------------------------------------------
+
+
+class RandomDataLoader:
+    def __init__(
```

Comment:
We should just provide BandomdataWithKVCache as built-in data-config so that we do not need to keep copying this in user script in every example.

Reply:
Indeed, I have that in mind also there is work item to track it. Let us finish the build-in kv cache data config later.